.circleci/config.yml

```diff
@@ -102,45 +57,32 @@ jobs:
           command: |
             brew install pyenv
 
-#      - restore_cache:
-#          keys:
-#            - pyenv-{{ .Environment.CIRCLE_JOB }}
-
       - run:
           name: install python
           command: |
             pyenv install $PYTHON -s
 
-#      - save_cache:
-#          paths:
-#            - ~/.pyenv
-#          key: pyenv-{{ .Environment.CIRCLE_JOB }}
-
-#      - restore_cache: *restore-cache-template
-
+      #TODO: add caching back in when demos have the same requirements and exist in the same venv
       - run:
           name: create virtualenv
           command: |
             eval "$(pyenv init -)"
             pyenv local $PYTHON
             python -m pip install virtualenv
-            python -m virtualenv env
 
       - run:
-          name: install dependencies
-          command: |
-            . env/bin/activate
-            pip install -r requirements_tests.txt
-
-#      - save_cache: *save-cache-template
-
-      - run: &run-tests-template
-          name: run tests
+          name: install dependencies and run tests
           command: |
-            . env/bin/activate
-            coverage run -m unittest discover
-            deactivate
-#      - run: *run-tests-template
+            demos=$(ls -d */ | grep -v "tests" | grep -v "circuit-fault-diagnosis" | sed 's/\///')
+            for demo in $demos
```

Comment:
See comment above. Never do this.

Instead, you can, if you're using extended globbing:
```
for demo in !(tests|circuit-fault-diagnosis); do
    ...
done
```

Reply:
It's very odd. CircleCI's python 3.7 bash does not seem to recognize ! as a negation.

```
circleci@35fa1179b352:~/repo$ for demo in !(tests|circuit-fault-diagnosis); do
bash: !: event not found
```
```
circleci@35fa1179b352:~/repo$ for demo in ! (tests|circuit-fault-diagnosis); do
bash: syntax error near unexpected token `('
```
Using `find` instead.

Reply:
Nevermind. Finally figured it out. Needed to turn on extglob in CircleCI

Reply:
Yes, `shopt -s extglob`. Sorry, wasn't explicit about that in my comment above (sometimes it's already enabled).

See `help shopt` and `man bash` (search for `extglob`).